### PR TITLE
MORL/D

### PR DESCRIPTION
**Pull request by ffelten**
This PR adds the code used for the MORL/D paper: https://arxiv.org/abs/2311.12495.

Results are hosted on ORLB.

### CODE REVIEW

File: morl_baselines/common/accrued_reward_buffer.py [hunk 1]
```diff
@@ -88,21 +88,17 @@ def cleanup(self):
         """Cleanup the buffer."""
         self.size, self.ptr = 0, 0
 
-    def get_all_data(self, max_samples=None, to_tensor=False, device=None):
+    def get_all_data(self, to_tensor=False, device=None):
         """Returns the whole buffer (with a specified maximum number of samples).
```
Comment (LucasAlegre) [inline morl_baselines/common/accrued_reward_buffer.py:92]: No longer with a specified number of samples :P

File: morl_baselines/multi_policy/gpi_pd/gpi_pd.py [hunk 2]
```diff
@@ -873,7 +877,7 @@ def train(
                     n_value = policy_evaluation_mo(self, eval_env, wcw, rep=num_eval_episodes_for_front)[3]
                     linear_support.add_solution(n_value, wcw)
 
-            if self.log:
+            if self.log and self.global_step % (eval_freq * 10) == 0:
```
Comment (LucasAlegre) [inline morl_baselines/multi_policy/gpi_pd/gpi_pd.py:880]: The idea here is to log multipolicy metrics after the end of every iteration. Why this change?

File: morl_baselines/multi_policy/gpi_pd/gpi_pd_continuous_action.py [hunk 3]
```diff
@@ -672,7 +674,7 @@ def train(
                     n_value = policy_evaluation_mo(self, eval_env, wcw, rep=num_eval_episodes_for_front)[3]
                     linear_support.add_solution(n_value, wcw)
 
-            if self.log:
+            if self.log and self.global_step % (eval_freq * 10) == 0:
```
Comment (LucasAlegre) [inline morl_baselines/multi_policy/gpi_pd/gpi_pd_continuous_action.py:677]: Same as above

**Review by LucasAlegre — CHANGES_REQUESTED** [review]

File: morl_baselines/multi_policy/gpi_pd/gpi_pd.py (hunk 2 above)
Comment (ffelten) [reply morl_baselines/multi_policy/gpi_pd/gpi_pd.py:880]: Because the evaluation is super slow in practice and I don't think we lose that much info when evaluating less in this case. I mean, it took 5 days to run GPI-LS for 3M steps IIRC 🫠

Comment (LucasAlegre) [reply morl_baselines/multi_policy/gpi_pd/gpi_pd.py:880]: I see, but this depends on the number of timesteps per iteration too, and the hardcoded 10 is misleading. I think this should be split into two parameters:

eval_freq: Inside an iteration, how often in time steps to do policy evaluation
x (idk a good name) : Eval the multi-policy metrics at every x iterations, in the outer loop.

Comment (ffelten) [reply morl_baselines/multi_policy/gpi_pd/gpi_pd.py:880]: I initially had `eval_freq_in_iter` and `eval_freq` to do that, which I merged through the *10 :p. But yeah, we can add a parameter.

Comment (ffelten) [reply morl_baselines/multi_policy/gpi_pd/gpi_pd.py]: I added an additional parameter `eval_mo_freq`

**Review by LucasAlegre — APPROVED** [review]
